statement_test.go

```diff
@@ -534,3 +536,60 @@ func TestBindJSON(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, string(jsonData), str)
 }
+
+func TestPrepareComplexQueryParameter(t *testing.T) {
+	db := openDbWrapper(t, ``)
+	defer closeDbWrapper(t, db)
+
+	ctx := context.Background()
+	createTable(t, db, `CREATE OR REPLACE TABLE arr_test(
+		arr_float float[2]
+	)`)
+
+	// Insert parameters.
+	_, err := db.ExecContext(ctx, `INSERT INTO arr_test VALUES (?)`,
+		[]float32{7, 1})
+	require.NoError(t, err)
+
+	prepared, err := db.Prepare(`SELECT list_value(?,?), list_cosine_distance(?,?)`)
```

Comment:
Do `list_value` and `list_cosine_distance` trigger the `tryBind...` code, or do we know their parameter types?